packages/itwin/property-grid/src/components/Header.tsx

```diff
@@ -6,32 +6,105 @@
 import "./Header.scss";
 import classnames from "classnames";
 import { SvgProgressBackwardCircular } from "@itwin/itwinui-icons-react";
-import { IconButton } from "@itwin/itwinui-react";
+import { IconButton, SearchBox  } from "@itwin/itwinui-react";
 import { PropertyGridManager } from "../PropertyGridManager";
 
-import type { PropsWithChildren } from "react";
+import { type PropsWithChildren, useEffect, useRef, useState } from "react";
+import type React from "react";
 
 /** @internal */
 export interface HeaderProps {
   className?: string;
   onBackButtonClick?: () => void;
+  titleItem?: React.ReactNode;
+  headerTools?: React.ReactNode;
+  setSearchInput?: (searchInput: string) => void;
 }
 
-/** @internal */
-export function Header({ className, children, onBackButtonClick }: PropsWithChildren<HeaderProps>) {
-  return <div className={classnames("property-grid-react-panel-header", className)}>
-    {
-      onBackButtonClick
-        ? <IconButton
-          styleType="borderless"
-          onClick={onBackButtonClick}
-          title={PropertyGridManager.translate("header.back")}
-          className="property-grid-react-header-back-button"
-        >
-          <SvgProgressBackwardCircular />
-        </IconButton>
-        : null
+export interface DebouncedSearchBoxProps {
+  setSearchInput: (value: string) => void;
+  onClose: () => void;
+  onOpen: () => void;
+  className: string;
+}
+
+export function DebouncedSearchBox({ setSearchInput, onClose, onOpen, className }: DebouncedSearchBoxProps){
+  const [inputValue, setInputValue] = useState<string>("");
+  const setSearchInputRef = useRef(setSearchInput);
+  setSearchInputRef.current = setSearchInput;
+
+  useEffect(() => {
+    if (!inputValue) {
+      setSearchInputRef.current("");
+      return;
     }
-    {children}
+
+    const timeoutId = setTimeout(() => {
+      setSearchInputRef.current(inputValue);
+    }, 25);
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, [inputValue]);
+
+  return <SearchBox
+    expandable
+    onCollapse={onClose}
+    onExpand={onOpen}
+    className={className}
+  >
+    <SearchBox.CollapsedState>
+      <SearchBox.ExpandButton
+        title="Expand Searchbar"
+      />
+    </SearchBox.CollapsedState>
+    <SearchBox.ExpandedState >
+      <SearchBox.Input
+        placeholder={"Search"}
+        onChange={(e) => setInputValue(e.currentTarget.value)}
+      />
+      <SearchBox.CollapseButton
+        title="Contract Searchbar"
```

Comment:
I think better title for this button would be `Collapse search box` or `Close search box`. Also should be localized.